crates/red_knot_python_semantic/resources/mdtest/stubs/class.md

```diff
@@ -15,3 +15,31 @@ class Bar(Foo[Bar]): ...
 reveal_type(Bar)  # revealed: Literal[Bar]
 reveal_type(Bar.__mro__)  # revealed: tuple[Literal[Bar], Unknown, Literal[object]]
 ```
+
+## Access to attributes declarated in stubs
+
+Unlike regular Python modules, stub files often omit the RHS in declarations, including in class
+scope. However, from the perspective of the type checker, we have to treat them as bindings too.
+That is, `symbol: type` is the same as `symbol: type = ...`.
+
+One implication of this is that we'll always treat symbols in class scope as either
+`class or instance` or `class only` (if `ClassVar` is used). We'll never infer a pure instance
+attribute from a stub.
```

Comment:
I'm not a huge fan of the term "class or instance" attribute. It doesn't appear in the typing spec, and conceptually I think it's the wrong model to have. The [spec](https://typing.readthedocs.io/en/latest/spec/class-compat.html) says that these are instance attributes that just have class-level defaults, and other than the fact that they can be accessed safely on the class object they behave like "pure" instance attributes in nearly every other way -- so I think that's the better way to think about them.

I'd suggest rewording this paragraph like so:

```suggestion
One implication of this is that we'll always treat symbols in class scope as safe to be
accessed from the class object itself. We'll never infer a "pure instance attribute" from
a stub.
```